aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1570,10 +1576,6 @@ Tensor reshape_symint(const Tensor& self, c10::SymIntArrayRef proposed_shape) {
     AT_ERROR("reshape is not implemented for sparse tensors");
   }
 
-  if (self.is_contiguous() && !self.is_mkldnn()) {
-    return self.view_symint(proposed_shape);
```

Comment:
This may break some unbacked SymInt tests, although after I added this branch we changed the strategy so maybe this is dead. Safer to do this separately.